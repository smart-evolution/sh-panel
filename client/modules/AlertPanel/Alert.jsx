import React from 'react';
import PropTypes from 'prop-types';

class Alert extends React.Component {
  constructor() {
    super();

    this.state = {
      show: false,
      hide: false,
    };
  }

  componentDidMount() {
    this.showTimeout = setTimeout(() => {
      this.setState({ show: true });
    }, 2000);

    this.hideTimeout = setTimeout(() => {
      this.setState({ hide: true });
    }, 4000);
  }

  componentWillUnmount() {
    if (this.showTimeout) {
      clearTimeout(this.showTimeout);
    }
    if (this.hideTimeout) {
      clearTimeout(this.hideTimeout);
    }
  }

  render() {
    const { type, children } = this.props;
    const { show, hide } = this.state;

    const showClass = show ? 'show' : '';
    const hideClass = hide ? 'hide' : '';
    const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;

    return <div className={classes}>{children}</div>;
  }
}

Alert.propTypes = {
  type: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Alert;
