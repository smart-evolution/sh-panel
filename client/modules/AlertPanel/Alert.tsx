import React from 'react';

type Props = {
  type: string;
  children: React.ReactNode;
};

type State = {
  show: boolean;
  hide: boolean;
};

class Alert extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      show: false,
      hide: false,
    };
  }

  componentDidMount() {
    // @ts-ignore - to be fixed
    this.showTimeout = setTimeout(() => {
      this.setState({ show: true });
    }, 2000);
    // @ts-ignore - to be fixed
    this.hideTimeout = setTimeout(() => {
      this.setState({ hide: true });
    }, 4000);
  }

  componentWillUnmount() {
    // @ts-ignore - to be fixed
    if (this.showTimeout) {
      // @ts-ignore - to be fixed
      clearTimeout(this.showTimeout);
    }
    // @ts-ignore - to be fixed
    if (this.hideTimeout) {
      // @ts-ignore - to be fixed
      clearTimeout(this.hideTimeout);
    }
  }

  render() {
    const { type, children } = this.props;
    const { show, hide } = this.state;

    const showClass = show ? 'show' : '';
    const hideClass = hide ? 'hide' : '';
    const classes = `alert-panel__alert alert-panel__alert--${type} ${showClass} ${hideClass}`;

    return <div className={classes}>{children}</div>;
  }
}

export default Alert;
