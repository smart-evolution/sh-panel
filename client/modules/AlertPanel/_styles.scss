.alert-panel {
  position: fixed;
  right: $spacing-large;
  bottom: $spacing-large;

  &__alert {
    @extend .c-alert;

    width: 350px;
    margin-bottom: $spacing-small;
    opacity: 1;
    transition: opacity 2s;

    &.show {
      opacity: 0;
    }

    &.hide {
      display: none;
    }

    &--info {
      @extend .c-alert--success;
    }

    &--error {
      @extend .c-alert--danger;
    }
  }
}
