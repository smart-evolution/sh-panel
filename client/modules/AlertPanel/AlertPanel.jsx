// @flow
import _ from 'lodash';
import React from 'react';
import Alert from './Alert';

type Props = {
  alerts: $ReadOnly<any>,
};

const AlertPanel = (props: Props) => {
  const { alerts } = props;

  return (
    <div className="alert-panel">
      {_.map(alerts, (alert, index) => {
        const { type, message } = alert;
        const key = `alert-${index}`;

        return (
          <Alert key={key} type={type}>
            {message}
          </Alert>
        );
      })}
    </div>
  );
};

export default AlertPanel;
