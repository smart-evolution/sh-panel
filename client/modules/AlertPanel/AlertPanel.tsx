// @flow
import _ from 'lodash';
import React from 'react';
import Alert from './Alert';

type Props = {
  alerts: any;
};

const AlertPanel = (props: Props) => {
  const { alerts } = props;

  return (
    <div className="alert-panel">
      {_.map(alerts, (alert) => {
        const { type, message } = alert;
        const key = `alert-${JSON.stringify(alert)}`;

        return (
          <Alert key={key} type={type}>
            {message}
          </Alert>
        );
      })}
    </div>
  );
};

export default AlertPanel;
