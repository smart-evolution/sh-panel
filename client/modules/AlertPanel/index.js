// @flow
import { connect } from 'react-redux';
import * as selectors from 'client/models/alerts/selectors';
import AlertPanel from './AlertPanel';

const mapStateToProps = state => {
  const alerts = selectors.getLimitedAlerts(state);

  return {
    alerts,
  };
};

export default connect(
  mapStateToProps,
  null
)(AlertPanel);
