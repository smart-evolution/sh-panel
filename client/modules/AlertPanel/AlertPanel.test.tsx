import React from 'react';
import { mount } from 'enzyme';
import AlertPanel from './AlertPanel';

describe('AlertPanel/AlertPanel', () => {
  it('should render correctly', () => {
    const alerts = [
      {
        type: 'type-class',
        message: 'Message 1',
      },
      {
        type: 'type-class',
        message: 'Message 2',
      },
      {
        type: 'type-class',
        message: 'Message 3',
      },
    ];

    const component = mount(<AlertPanel alerts={alerts} />);

    expect(component).toMatchSnapshot();
  });
});
