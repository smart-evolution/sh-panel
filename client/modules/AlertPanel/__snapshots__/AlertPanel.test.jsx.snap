// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertPanel/AlertPanel should render correctly 1`] = `
<AlertPanel
  alerts={
    Array [
      Object {
        "message": "Message 1",
        "type": "type-class",
      },
      Object {
        "message": "Message 2",
        "type": "type-class",
      },
      Object {
        "message": "Message 3",
        "type": "type-class",
      },
    ]
  }
>
  <div
    className="alert-panel"
  >
    <Alert
      key="alert-0"
      type="type-class"
    >
      <div
        className="alert-panel__alert alert-panel__alert--type-class  "
      >
        Message 1
      </div>
    </Alert>
    <Alert
      key="alert-1"
      type="type-class"
    >
      <div
        className="alert-panel__alert alert-panel__alert--type-class  "
      >
        Message 2
      </div>
    </Alert>
    <Alert
      key="alert-2"
      type="type-class"
    >
      <div
        className="alert-panel__alert alert-panel__alert--type-class  "
      >
        Message 3
      </div>
    </Alert>
  </div>
</AlertPanel>
`;
